prow/gcpproxy-api-regression.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+
+# Copyright 2019 Google LLC
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Presubmit script triggered by Prow.
+
+# Fail on any error.
+set -eo pipefail
+
+WD=$(dirname "$0")
+WD=$(cd "$WD";
+pwd)
+ROOT=$(dirname "$WD")
+export PATH=$PATH:$GOPATH/bin
+
+
+gcloud config set core/project cloudesf-testing
+gcloud auth activate-service-account \
+  --key-file="${GOOGLE_APPLICATION_CREDENTIALS}"
+
+ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd)"
+cd "${ROOT}"
+. ${ROOT}/scripts/all-utilities.sh || { echo 'Cannot load Bash utilities';
+exit 1; }
+. ${ROOT}/tests/e2e/scripts/prow-utilities.sh || { echo 'Cannot load Bash utilities';
+exit 1; }
+
+
+wait_apiproxy_image
+
+echo '======================================================='
+echo '================ Download latest envoy ================'
+echo '======================================================='
+download_envoy_binary
+chmod +x ${ROOT}/bin/envoy
+
+# keep the current version
+VERSION=$(cat ${ROOT}/VERSION)
+
+# checkout to last releasd commit
+SHA=$(git rev-list -n 1 $(git describe --abbrev=0))
```

Comment:
Hmm, why checkout to a released commit? Why not just checkout to the latest commit on master (the diff would be all the commits in the individual PR).

The problem with checking out to a released commit is: As soon as we submit one PR that breaks the test on purpose, then all the following PRs will keep failing until a new release is made. That doesn't sound very helpful then.

Also if we increment the API version, every single PR afterwords will fail until we make a new release.

Better to just test against master. Each PR will be responsible for making sure they don't break the API version. If the previous PR didn't break the version, then no need to checkout to a older commit than it.

Reply:
Ack, good point.

Reply:
Done.